ls1intum/Artemis
# [Enhancement] Reduce code duplication in for-dashboard call

### Checklist

### Motivation and Context


When an assessment is just saved and not submitted, the result has not completion date yet. This lead to a NPE in when updating the latestSubmission according to the completion date of the submissions result. 

### Description

- Assure that only manual results that are rated and have a completion date are taken into account 
- Reduced duplicated code, by using some more checks in `ExerciseService.java`

### Steps for Testing

More focus on the code review

1. Log in to Artemis
2. Navigate to Course Administration
3. Assess a programming exercise submission and just save
4. Go to Course Overview and see that the for-dashboard throws no error
5. Play a bit around everything should work as expected (e.g. Assessed results are displayed after assessment due date)
See Instructions `Publishing of results before assessment due date and after assessment due date (student)` in #2137

### Test Coverage

Nothing changed, just reduced code duplication

### Screenshots
